conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptions.java

```diff
@@ -89,27 +93,12 @@ private static void serviceException(HttpServerExchange exchange, ServiceExcepti
 
     private static void qosException(HttpServerExchange exchange, QosException qosException) {
         qosException.accept(QOS_EXCEPTION_HEADERS).accept(exchange);
-        if (qosExceptionHasAdditionalMetadata(qosException)) {
-            log.info("quality-of-service intervention", qosException);
-        } else {
-            log.debug("quality-of-service intervention", qosException);
-        }
-        writeResponse(exchange, Optional.empty(), qosException.accept(QOS_EXCEPTION_STATUS_CODE));
-    }
 
-    /** Returns  true if {@link QosException} provides additional metadata and should be logged at {@code info}. */
-    private static boolean qosExceptionHasAdditionalMetadata(QosException qosException) {
-        try {
-            if (qosException.getCause() != null) {
-                return true;
-            }
-            Throwable[] suppressed = qosException.getSuppressed();
-            return suppressed != null && suppressed.length > 1;
-        } catch (Throwable t) {
-            // Fallback in case of unexpected bytecode manipulation (getCause or getSuppressed throws due to
-            // poor instrumentation/mocks)
-            return true;
+        if (qosLoggingRateLimiter.tryAcquire()) {
+            log.info("Quality-of-Service error handling request", qosException);
```

Comment:
Updated this log message to be a bit more human-readable and more closely match the other log messages used here.